### PR TITLE
My Jetpack: Install products from interstitial page

### PR DESCRIPTION
**Pull request by oskosk**
Adds mechanism for sideloading product plugins when clicking the _Add Product_ button on interstitial pages.

**I am not so sure I like this flow now that I've seen it working here.**

* Makes Boost and Backup be installed from the interstitial pages.

#### Changes proposed in this Pull Request:

* Adds button component with spinner next to product-detail-card component. Jetpack's ActionButton is not styled properly as to be useful here rapidly
* Replaces usage of regular Button for this new ProductDetailButton

#### Does this pull request change what data or activity we track or use?
no

#### Test Scan acquisition flow
1. Test that the scan interstitial leads directly to the checkout page for the Scan and Security products.

#### Testing Boost sideloading

2. Remove the boost plugin
3. Visit My Jetpack.
4. Click Add Boost link. Expect to see the interstitial page for Boost
5. Click Add boost button on the interstitial. Expect to see a spinner
6. Expect to get reidrected to the My Jetpack page again

https://user-images.githubusercontent.com/746152/153989964-d2d8c7d1-7f02-4a34-aef4-c7b7e97bf961.mov

#### Testing Backup sideloading

2. Remove the Backup plugin
3. Temporarily edit line 27 of file `/projects/packages/my-jetpack/src/products/class-hybrid-product.php` to be
  ```
  return static::is_plugin_active();// || static::is_jetpack_plugin_active();

  ```
3. Visit My Jetpack.
4. Click Add Backup link. Expect to see the interstitial page for Backup
5. Click Add Backup button on the interstitial. Expect to see a spinner
6. Expect to get redrected to the Checkout page. The current flow for Backup is failing so expect to see a failure there on Calypso

https://user-images.githubusercontent.com/746152/153991065-a5000d11-5af1-4653-b7a9-3295941a253e.mov

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx [hunk 1]
```diff
@@ -40,7 +46,39 @@ export default function ProductInterstitial( { slug, children = null } ) {
...
 
+	const clickHandler = useCallback( () => {
+		if ( installsPlugin ) {
+			activate()
+				.then( () => {
+					afterInstallation( isFree );
+				} )
+				.catch( () => {
+					afterInstallation( isFree );
+				} );
```
Comment (retrofox) [inline projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx:79]: minor suggestion.
Maybe you can use [.finally()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally) here to void duplicate the same call for `then()` and `catch()`.

File: projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx [hunk 2]
```diff
@@ -78,7 +120,7 @@ export function AntiSpamInterstitial() {
  */
 export function BackupInterstitial() {
 	return (
-		<ProductInterstitial slug="backup">
+		<ProductInterstitial slug="backup" installsPlugin={ true }>
```
Comment (retrofox) [inline projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx:123]: Just a note: I think it shouhd be handled at the hook/data level. If so, we can iterate in follow-up PRs.

File: projects/packages/my-jetpack/_inc/components/product-detail-card/index.jsx [hunk 3]
```diff
@@ -70,11 +70,12 @@ function Price( { value, currency, isOld } ) {
  *
  * @param {object} props                    - Component props.
  * @param {string} props.slug               - Product slug
+ * @param {Function} props.onClick          - Callback for Call To Action button click
```
Comment (retrofox) [inline projects/packages/my-jetpack/_inc/components/product-detail-card/index.jsx:73]: minor: `onClick` name is confusing for the `<ProductDetail />`. Maybe `onAdd()` is more descriptive.

**Review by retrofox — APPROVED** [review]
LGTM.
Again, I think we should try to handle the Product activation at (which includes installing, checkout process, etc) at the data level.

**Review by leogermani — APPROVED** [review]

**Pull request merged**